main.tf

```diff
@@ -0,0 +1,277 @@
+  subnets            = var.subnets
+  internal           = var.internal
+  load_balancer_type = var.load_balancer_type
+
+  security_group_rules = {
+    ingress_all = {
+      type        = "ingress"
+      from_port   = 0
+      to_port     = 0
+      protocol    = "-1"
+      cidr_blocks = ["0.0.0.0/0"]
+    }
+    egress_all = {
+      type        = "egress"
+      from_port   = 0
+      to_port     = 0
+      protocol    = "-1"
+      cidr_blocks = ["0.0.0.0/0"]
+    }
+  }
+
+  target_groups = [
+    {
+      name             = "${module.this.id}-${var.port_gateway}"
+      backend_protocol = "TCP"
+      backend_port     = var.port_gateway
+    },
+    {
+      name             = "${module.this.id}-${var.port_metadata}"
+      backend_protocol = "TCP"
+      backend_port     = var.port_metadata
+    },
+    {
+      name             = "${module.this.id}-${var.port_health}"
+      backend_protocol = "TCP"
+      backend_port     = var.port_health
+    },
+    {
+      name             = "${module.this.id}-80"
+      backend_protocol = "TCP"
+      backend_port     = 80
+    }
+  ]
+
+  http_tcp_listeners = [
+    {
+      port               = var.port_gateway
+      protocol           = "TCP"
+      target_group_index = 0
+    },
+    {
+      port               = var.port_metadata
+      protocol           = "TCP"
+      target_group_index = 1
+    },
+    {
+      port               = var.port_health
+      protocol           = "TCP"
+      target_group_index = 2
+    },
+    {
+      port               = 80
+      protocol           = "TCP"
+      target_group_index = 3
+    },
+  ]
+
+  tags = module.this.tags
+}
+
+module "ecs_label" {
+  source  = "cloudposse/label/null"
+  version = "0.25.0"
+
+  label_order = var.label_orders.ecs
+
+  context = module.this.context
+}
+
+resource "aws_cloudwatch_log_group" "default" {
+  count = var.cloudwatch_log_group_enabled ? 1 : 0
+
+  name              = module.this.id
+  tags              = module.this.tags
+  retention_in_days = var.log_retention_in_days
+}
+
+module "container_definition" {
+  source  = "cloudposse/ecs-container-definition/aws"
+  version = "0.58.1"
+
+  container_name  = module.ecs_label.id
+  container_image = "traefik:${var.container_image_tag}"
+  port_mappings = [
+    {
+      containerPort = var.port_gateway
+      hostPort      = 0
+      protocol      = "tcp"
+    },
+    {
+      containerPort = var.port_health
+      hostPort      = 0
+      protocol      = "tcp"
+    },
+    {
+      containerPort = var.port_metadata
+      hostPort      = 0
+      protocol      = "tcp"
+    },
+    {
+      containerPort = var.port_traefik
+      hostPort      = 0
+      protocol      = "tcp"
+    },
+  ]
+
+  command = [
+    "--entrypoints.gateway.address=:${var.port_gateway}/tcp",
+    "--entrypoints.health.address=:${var.port_health}/tcp",
+    "--entrypoints.metadata.address=:${var.port_metadata}/tcp",
+    "--entrypoints.traefik.address=:${var.port_traefik}/tcp",
```

Comment:
Could we simply define var.ports as set(string) which we then iterate over to generate the arguments and above the port mappings? That would be a lot cleaner, because module callers can then simply add/replace the ports when needed

Reply:
Maybe we can create a follow-up for that as it would be quite complicated

Reply:
would be nice if you created a follow up issue, however it does not have any high prio right now